community/detectors/joomla_cve_2023_23752/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202323752/Cve202323752VulnDetectorTest.java

```diff
@@ -0,0 +1,192 @@
+            return new MockResponse()
+                .addHeader("Content-Type", "application/json; charset=utf-8")
+                .setBody(LeakedDataJsonSample)
+                .setResponseCode(HttpStatus.OK.code());
+          }
+          return new MockResponse().setResponseCode(404);
+        }
+      };
+
+  @Test
+  public void DetectCVE202323752() {
+    mockWebServer.setDispatcher(dispatcher);
+    DetectionReportList mockWebServerDetectionReports =
+        detector.detect(targetInfo, ImmutableList.of(joomlaService));
+
+    // all we need to check is Detection Status But I think it is very hard to set the
+    // addAdditionalDetails , so I add the original Report addAdditionalDetails here in
+    // expected and then check for AdditionalDetails in another assert
+    DetectionReport expectedDetectionReport =
+        DetectionReport.newBuilder()
+            .setTargetInfo(targetInfo)
+            .setNetworkService(joomlaService)
+            .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+            .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+            .setVulnerability(
+                Vulnerability.newBuilder()
+                    .setMainId(
+                        VulnerabilityId.newBuilder()
+                            .setPublisher("TSUNAMI_COMMUNITY")
+                            .setValue("CVE_2023_23752"))
+                    .setSeverity(Severity.HIGH)
+                    .setTitle("Joomla unauthorized access to webservice endpoints")
+                    .setDescription(
+                        "CVE-2023-23752: An improper access check allows unauthorized access to"
+                            + " webservice endpoints. attacker can get the host address "
+                            + "and username and password of the configured joomla database.")
+                    .setRecommendation("Upgrade Joomla to 4.2.8 and above versions.")
+                    .addAdditionalDetails(
+                        AdditionalDetail.newBuilder()
+                            .setTextData(TextData.newBuilder().setText(LeakedDataJsonSample))))
+            .build();
+
+    // Vulnerable to CVE202323752
+    assertThat(mockWebServerDetectionReports.getDetectionReportsList())
+        .containsExactly(expectedDetectionReport);
+  }
+
+  @Test
+  public void detect_whenNotVulnerable_returnsNoVulnerability() {
+    mockWebServer.url("/notexistpath");
+    MockResponse response =
+        new MockResponse()
+            .addHeader("Content-Type", "application/json; charset=utf-8")
+            .setBody("NotExistDetectionString")
+            .setResponseCode(200);
+    mockWebServer.enqueue(response);
+
+    DetectionReportList mockWebServerDetectionReports =
+        detector.detect(targetInfo, ImmutableList.of(joomlaService));
+    assert (mockWebServerDetectionReports.getDetectionReportsList().isEmpty());
```

Comment:
```suggestion
    assertThat(mockWebServerDetectionReports.getDetectionReportsList()).isEmpty();
```
Use `assertThat` instead of `assert`